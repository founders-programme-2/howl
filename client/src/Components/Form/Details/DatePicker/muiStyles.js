// TODO: Refactor media query to take in constant variable for size

const styles = () => ({
  helperText: {
    fontFamily: 'Roboto',
    fontVariant: 'none',
    fontSize: 14,
  },
  grid: {
    width: '30%',
    display: 'flex',
    justifyContent: 'center',
    marginLeft: '5%',
    marginRight: '5%',
    background: 'white',
    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
    borderRadius: 4,
    padding: '2%',
    '@media (max-width: 500px)': {
      width: '80%',
    },
  },
});

export default styles;
