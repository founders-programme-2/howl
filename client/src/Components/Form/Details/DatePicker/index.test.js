import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from './index';

test('Natural', () => {
  let state = {};
  const props = {
    selectedDate: new Date(),
    handleDateChange: date => {
      const options = {
        year: 'numeric',
        month: 'long',
      };
      const dateModified = date.toLocaleDateString('en-US', options);
      const splitted = dateModified.split(' ');
      state = {
        ...state,
        selectedDate: dateModified,
        month: splitted[0],
        year: splitted[1],
      };
    },
  };
  const datePicker = renderer.create(<DatePicker {...props} />).toJSON();
  expect(datePicker).toMatchSnapshot();
});
