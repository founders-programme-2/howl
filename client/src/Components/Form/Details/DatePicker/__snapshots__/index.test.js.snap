// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Natural 1`] = `
<div
  className="MuiGrid-root DateFun-grid-2"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
      data-shrink={true}
    >
      Year and Month
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-159 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-160"
          style={
            Object {
              "width": 0,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="May 2019"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <p
      className="MuiFormHelperText-root DateFun-helperText-1 MuiFormHelperText-contained MuiFormHelperText-filled"
    >
      When did this happen?
    </p>
  </div>
</div>
`;
